File: TP1/arquivosJAVA/PalindromoR.java
import java.util.Scanner;

public class PalindromoR {

    static boolean teste(String palavra, int esquerda, int direita) {

        if (esquerda > direita) {
            return true;
        }

        if (palavra.charAt(esquerda) == palavra.charAt(direita)) {

            teste(palavra, esquerda + 1, direita - 1);

        } else {
            return false;
        }
        return true;
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        String palavra;

        while (true) {
            palavra = sc.nextLine();
            int esquerda = 0;
            int direita = palavra.length() - 1;

            if (palavra.equals("FIM")) {
                sc.close();
                System.exit(0);
            }

            if (teste(palavra, esquerda, direita) == true) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }

        }
    }
}

File: TP1/arquivosC/PalindromoR.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <ctype.h>

bool palindromo(char palavra[], int esquerda,int direita){



    if(esquerda>direita){
        return false;
    }


        if(palavra[esquerda]!=palavra[direita]){
            return false;
        }else{
            palindromo(palavra,esquerda+1,direita-1);
        }


    return true;
}



int main(){

    char palavra[1000];

    while(1){

    scanf(" %[^\n]", palavra);
    int tamanho;
    tamanho=strlen(palavra);

    int esquerda=0;
    int direita=tamanho-1;

    if (strcmp(palavra, "FIM") == 0) {
            return 0;
        }

    printf(palindromo(palavra,esquerda,direita)? "SIM\n":"NAO\n");

    }


    return 0;
}